language-plutus-core/src/Language/PlutusCore/Error.hs

```diff
@@ -95,15 +95,15 @@ data InternalTypeError uni ann
 makeClassyPrisms ''InternalTypeError
 
 data TypeError uni ann
-    = KindMismatch ann (Type TyName uni ()) (Kind ()) (Kind ())
+    = KindMismatch ann (Type TyName uni ()) (Kind ())  (Kind ())
     | TypeMismatch ann
         (Term TyName Name uni ())
         (Type TyName uni ())
         (Normalized (Type TyName uni ()))
     | UnknownDynamicBuiltinName ann UnknownDynamicBuiltinNameError
     | InternalTypeErrorE ann (InternalTypeError uni ann)
-    | FreeTypeVariableE (TyName ann)
-    | FreeVariableE (Name ann)
+    | FreeTypeVariableE TyName
+    | FreeVariableE Name
```

Comment:
So this does mean we potentially lose information about where the variable was bound in these errors, but I think that's okay. We could add it in if we want it.

Reply:
The other constructors do have annotations, so I think they should be added here as well.